remoting/src/main/java/org/apache/rocketmq/remoting/netty/NettyRemotingServer.java

```diff
@@ -169,6 +188,7 @@ public void initChannel(SocketChannel ch) throws Exception {
                             new NettyDecoder(),
                             new IdleStateHandler(0, 0, nettyServerConfig.getServerChannelMaxIdleTimeSeconds()),
                             new NettyConnectManageHandler(),
+                            new StartTlsHandler(),
```

Comment:
Nice catch. We do can simplify this process by detecting handshake per https://tools.ietf.org/html/rfc5246#section-6.2.1